src/test/common/test_random.cc

```diff
@@ -0,0 +1,205 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+*/
+
+#include <sstream>
+
+#include "include/random.h"
+
+#include "gtest/gtest.h"
+
+// Helper to see if calls compile with various types:
+template <typename T>
+T type_check_ok(const T min, const T max)
+{
+  return ceph::util::generate_random_number(min, max);
+}
+
+/* Help wrangle "unused variable" warnings: */
+template <typename X>
+void swallow_values(const X x)
+{
+ static_cast<void>(x);
+}
+
+template <typename X, typename ...XS>
+void swallow_values(const X x, const XS... xs)
+{
+ swallow_values(x), swallow_values(xs...);
+}
+
+/* This test needs to run before any other calls into the rng are made, because
+internally it uses thread_local storage; gtest runs these in one thread, so the
+expected false-state can't really be known. */
```

Comment:
for tests that depend on tls, you should just wrap them in a lambda and run on a separate thread:
```c++
  std::thread([] {
    ASSERT_EQ(false, ceph::util::rng_is_seeded());
    ...
  }).join();
```
that way, we don't have to rely on test ordering, and arguments like `--gtest_repeat` will continue to work